### PR TITLE
Remove unused code in container manager

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The `unsupportedPodContainerManager` was introduced by https://github.com/kubernetes/kubernetes/pull/30475/files#diff-1f52d86b83eb91dcc3c28a9a260df6235fb63544ed1afa2ea62e161aba8cfc94.
But it has been not used since https://github.com/kubernetes/kubernetes/pull/56337/files#diff-1f52d86b83eb91dcc3c28a9a260df6235fb63544ed1afa2ea62e161aba8cfc94.
This PR is a followup of #56337 to remove the unused code in container manager.

#### Which issue(s) this PR fixes:

Ref #56337

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/cm/pod_container_manager_unsupported.go [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-limitations under the License.
-*/
-
-package cm
-
-type unsupportedPodContainerManager struct {
-	podContainerManagerStub
-}
-
-var _ PodContainerManager = &unsupportedPodContainerManager{}
```
Comment (bart0sh) [inline pkg/kubelet/cm/pod_container_manager_unsupported.go:26]: /lgtm

**Comment by bart0sh** [comment]
/remove-lifecycle stale

/cc @mrunalp @derekwaynecarr @dchen1107 
for approval

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**